### ISSUE TITLE
Flaky test: AppCanHandleClientAbortingConnectionMidResponse

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/KestrelHttpServer#2581]
This test [fails](http://aspnetci/viewLog.html?buildId=469532&buildTypeId=Lite_KestrelStressTest) occasionally with the following error:
```
Moq.MockException : 
Expected invocation on the mock once, but was 0 times: t => t.ConnectionStop(It.IsAny<String>())
No setups configured.
Performed invocations:
KestrelTrace.IsEnabled(Critical)
KestrelTrace.BeginScope<ConnectionLogScope>([[ConnectionId, 0HLDRBFHJGCHT]])
KestrelTrace.ConnectionStart("0HLDRBFHJGCHT")
KestrelTrace.ConnectionDisconnect("0HLDRBFHJGCHT")
   at Moq.Mock.ThrowVerifyException(MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, Expression expression, Times times, Int32 callCount)
   at Moq.Mock.VerifyCalls(Interceptor targetInterceptor, MethodCall expected, Expression expression, Times times)
   at Moq.Mock.Verify[T](Mock`1 mock, Expression`1 expression, Times times, String failMessage)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ResponseTests.AppCanHandleClientAbortingConnectionMidResponse(ListenOptions listenOptions) in /_/test/Kestrel.FunctionalTests/ResponseTests.cs:line 2504
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by mikeharder** [comment]
Another hit: http://aspnetci/viewLog.html?buildId=471422&tab=buildResultsDiv&buildTypeId=Lite_KestrelStressTest#testNameId747127487641677752

**Comment by mikeharder** [comment]
Another hit: http://aspnetci/viewLog.html?buildId=473129&tab=buildResultsDiv&buildTypeId=Lite_KestrelStressTest

@halter73 : Should this be fixed by https://github.com/aspnet/KestrelHttpServer/issues/2565?

**Comment by halter73** [comment]
This was fixed by #2574.
